webui/src/lib/onboarding/repoOnboardingService.tsx

```diff
@@ -0,0 +1,136 @@
+/* eslint-disable react/jsx-no-target-blank */
+import React from "react";
+import { OnBoardingSteps } from "./types";
+import { NavigateFunction } from "react-router-dom";
+import { branches, retention, branchProtectionRules, objects } from "../api";
+
+const IMPORT_BRANCH_REGEX = /^_(.*)_imported$/;
+const HOOKS_PATH = "_lakefs_actions/";
+const SAMPLE_REPO_FILE_NAME = "lakes.parquet";
+
+// generate a random file name to avoid collisions for the upload preflight check
+const generateRandomFileName = (): string => {
+  const randomString = Math.random().toString(36).substring(2, 15);
+  return `sample_${randomString}.yaml`;
+};
+
+export const isSampleRepo = async (
+  repo: string,
+  defaultBranch: string,
+): Promise<boolean> => {
+  const objectList = await objects.list(repo, defaultBranch, "");
```

Comment:
I still don't understand how this works.  `objects.list` paginates, so you have no way of knowing whether SAMPLE_REPO_FILE_NAME does not appear in the list because it's not there or because it will appear on a subsequent page.
Why not use headObject or any other test for object existence?

Reply:
Replaced `objects.list` with `objects.head`